metrics/prometheus/client.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 Mark Nevill. All Rights Reserved.
+// See LICENSE for licensing terms.
+
+package http_prometheus
+
+import (
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/mwitkow/go-httpwares/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	clientStarted = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "http_client_started_requests_total",
+			Help: "Count of started requests.",
+		},
+		[]string{"name", "host", "path", "method"},
+	)
+	clientCompleted = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "http_client_completed_requests_total",
+			Help: "Count of completed requests.",
+		},
+		[]string{"name", "host", "path", "method", "status"},
+	)
+	clientLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "http_client_completed_latency_seconds",
+			Help:    "Latency of completed requests.",
+			Buckets: []float64{.01, .03, .1, .3, 1, 3, 10, 30, 100, 300},
```

Comment:
hm, this should be allowed to be customized, isn't?

Reply:
having the metrics customisable adds complexity as you need to register separate metrics with separate names etc. I've tried to pick buckets that will work quite generally. We can revisit if we really need it.